hedera-node/src/main/java/com/hedera/services/txns/token/process/Creation.java

```diff
@@ -120,9 +120,15 @@ public void doProvisionallyWith(long now, TokenModelFactory modelFactory, NewRel
 		provisionalToken = modelFactory.createFrom(provisionalId, op, treasury, autoRenew, now);
 		provisionalToken.getCustomFees().forEach(fee ->
 				fee.validateAndFinalizeWith(provisionalToken, accountStore, tokenStore));
-		newRels = listing.listFrom(provisionalToken, dynamicProperties.maxTokensPerAccount());
+		final var hasExistingAssociations = treasury.getLastAssociatedToken().value() != 0;
+		newRels = listing.listFrom(provisionalToken, tokenStore);
 		if (op.getInitialSupply() > 0) {
-			provisionalToken.mint(newRels.get(0), op.getInitialSupply(), true);
+			/* The new treasury relation will either be at 0 [treasury has no other relationships] index or
```

Comment:
We may want to rename `newRels` to `newAndUpdatedRels`; and elaborate a bit more here,
```
// When we created the new relationship for a treasury that already had a last-added relationship, 
// we had to _first_ update the prev pointer on that relationship; so it will come first in the list of
// newAndUpdatedRels---and the new relationship with this token will come second
```